arduino-ide-extension/src/common/protocol/boards-service.ts

```diff
@@ -647,3 +648,127 @@ export function sanitizeFqbn(fqbn: string | undefined): string | undefined {
   const [vendor, arch, id] = fqbn.split(':');
   return `${vendor}:${arch}:${id}`;
 }
+
+export interface BoardConfig {
+  selectedBoard?: Board;
+  selectedPort?: Port;
+}
+
+export interface BoardInfo {
+  /**
+   * Board name. Could be `'Unknown board`'.
+   */
+  BN: string;
+  /**
+   * Vendor ID.
+   */
+  VID: string;
+  /**
+   * Product ID.
+   */
+  PID: string;
+  /**
+   * Serial number.
+   */
+  SN: string;
+}
+
+export const selectBoardForInfo = nls.localize(
+  'arduino/board/selectBoardForInfo',
+  'Please select a board to obtain board info.'
+);
+export function platformMissing(
+  name: string
+): string | BoardInfo | PromiseLike<string | BoardInfo> {
+  return nls.localize(
+    'arduino/board/platformMissing',
+    "The platform for the selected '{0}' board is not installed.",
+    name
+  );
+}
+export const selectPortForInfo = nls.localize(
+  'arduino/board/selectPortForInfo',
+  'Please select a port to obtain board info.'
+);
+export const nonSerialPort = nls.localize(
+  'arduino/board/nonSerialPort',
+  "Non-serial port, can't obtain info."
+);
+export const noNativeSerialPort = nls.localize(
+  'arduino/board/noNativeSerialPort',
+  "Native serial port, can't obtain info."
+);
+export const unknownBoard = nls.localize(
+  'arduino/board/unknownBoard',
+  'Unknown board'
+);
+
+/**
+ * The returned promise resolves to a `BoardInfo` if available to show in the UI or an info message explaining why showing the board info is not possible.
+ */
+export async function getBoardInfo(
+  boardsConfig: BoardConfig,
+  availablePorts: MaybePromise<AvailablePorts>
+): Promise<BoardInfo | string> {
+  const { selectedBoard, selectedPort } = boardsConfig;
+  if (!selectedBoard) {
+    return selectBoardForInfo;
+  }
```

Comment:
Is there a reason having a board selected is required? I questioned this in my previous review, which gives more info:

https://github.com/arduino/arduino-ide/pull/1803#pullrequestreview-1252844000

Reply:
In IDE2, selecting a port without a board is impossible.

Reply:
It can be done by following the instructions I provide at https://github.com/arduino/arduino-ide/pull/1803#pullrequestreview-1252844000:

![image](https://user-images.githubusercontent.com/8572152/216622670-7674a9eb-9f6b-4c86-8018-0e5417f3c196.png)

Reply:
Yes, you're right. Thanks!

Reply:
It's nice that you take the time to look into the code. You're a great help to me!

I have updated the PR as you asked me to. Please check again. Thanks!